### PR TITLE
Various fixes that improve the overall experience when working with the driver from Excel, Power BI etc.

### PR DESCRIPTION
**Pull request by traceon**
Extend SQL_CVT_* vs. SQL_CONVERT_* flags mapping
Fix compilation in windows
Fix null-termination logic in output strings
Fix SQLGetTypeInfo prototype
Fix SQLColAttribute() value reporting
Report SQL_OUTER_JOINS as supported
Fix DATA_TYPE in column description
Add LowCardinality detection to the type parser
Other cleanups

### CODE REVIEW

File: driver/api/impl/info.cpp [hunk 1]
```diff
@@ -148,31 +152,31 @@ SQLRETURN SQL_API EXPORTED_FUNCTION_MAYBE_W(SQLGetInfo)(
...
+            CASE_FALLTHROUGH(SQL_CONVERT_DATE)
             CASE_FALLTHROUGH(SQL_CONVERT_TIME)
             CASE_FALLTHROUGH(SQL_CONVERT_TIMESTAMP)
-            CASE_FALLTHROUGH(SQL_CONVERT_TINYINT)
-            CASE_FALLTHROUGH(SQL_CONVERT_VARBINARY)
-
-            CASE_NUM(SQL_CONVERT_VARCHAR, SQLUINTEGER, mask_SQL_CONVERT_VARCHAR)
+            CASE_FALLTHROUGH(SQL_CONVERT_INTERVAL_YEAR_MONTH)
+            CASE_FALLTHROUGH(SQL_CONVERT_INTERVAL_DAY_TIME)
+            CASE_NUM(SQL_CONVERT_GUID, SQLUINTEGER, mask_common_CONVERT_dest)
```
[the remark this answers is not in the document]
Comment (traceon) [reply driver/api/impl/info.cpp]: The former code assumed pretty much all possibilities, just missed some formal types like wide-char strings. I just added them.

File: driver/api/odbc.cpp [hunk 2]
```diff
@@ -908,111 +907,69 @@ SQLRETURN SQL_API EXPORTED_FUNCTION_MAYBE_W(SQLColAttribute)(
...
+            CASE_FIELD_NUM(SQL_DESC_LENGTH, (!type_info.isBufferType() ? 0 :
+                std::min<int32_t>(statement.getParent().stringmaxlength, (column_info.fixed_size ? column_info.fixed_size : column_info.display_size))
+            ));
+            CASE_FIELD_STR(SQL_DESC_LITERAL_PREFIX, "");
+            CASE_FIELD_STR(SQL_DESC_LITERAL_SUFFIX, "");
+            CASE_FIELD_STR(SQL_DESC_LOCAL_TYPE_NAME, "");
+            CASE_FIELD_STR(SQL_DESC_NAME, column_info.name);
+            CASE_FIELD_NUM(SQL_DESC_NULLABLE, (column_info.is_nullable ? SQL_NULLABLE : SQL_NO_NULLS));
+            CASE_FIELD_NUM(SQL_DESC_NUM_PREC_RADIX, (type_info.isIntegerType() ? 10 : 0));
+            CASE_FIELD_NUM(SQL_DESC_OCTET_LENGTH, (!type_info.isBufferType() ? type_info.octet_length :
```
[the remark this answers is not in the document]
Comment (traceon) [reply driver/api/odbc.cpp]: I don't think that's a good idea (with `if`'s) - the former code hid some of the issues due to that. The macro'ed `case` style is used everywhere else throughout the code currently, so if there is a better substitution to that, we'll need to change all such places.

File: driver/utils/utils.h [hunk 3]
```diff
@@ -220,14 +220,14 @@ inline SQLRETURN fillOutputString(
...
             if (out_value)
-                reinterpret_cast<CharType *>(out_value)[out_value_max_length_in_symbols] = 0;
-            rc = SQL_SUCCESS_WITH_INFO;
+                reinterpret_cast<CharType *>(out_value)[converted_length_in_symbols] = CharType{};
         }
         else {
             if (out_value)
-                reinterpret_cast<CharType *>(out_value)[converted_length_in_symbols] = 0;
+                reinterpret_cast<CharType *>(out_value)[out_value_max_length_in_symbols - 1] = CharType{};
+            rc = SQL_SUCCESS_WITH_INFO;
```
[the remark this answers is not in the document]
Comment (traceon) [reply driver/utils/utils.h]: I'll add that.

File: driver/utils/utils.h [hunk 4]
```diff
@@ -220,14 +220,14 @@ inline SQLRETURN fillOutputString(
...
+        if (converted_length_in_symbols < out_value_max_length_in_symbols) {
             if (out_value)
-                reinterpret_cast<CharType *>(out_value)[out_value_max_length_in_symbols] = 0;
-            rc = SQL_SUCCESS_WITH_INFO;
+                reinterpret_cast<CharType *>(out_value)[converted_length_in_symbols] = CharType{};
         }
         else {
             if (out_value)
-                reinterpret_cast<CharType *>(out_value)[converted_length_in_symbols] = 0;
+                reinterpret_cast<CharType *>(out_value)[out_value_max_length_in_symbols - 1] = CharType{};
```
[the remark this answers is not in the document]
Comment (traceon) [reply driver/utils/utils.h]: I'll add that.

File: driver/api/odbc.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (Enmk) [reply driver/api/odbc.cpp]: It is not about the macro, it is about three levels of trenary operator, the one used to compute the ‘VALUE’.

[the remark this answers is not in the document]
Comment (traceon) [reply driver/api/odbc.cpp]: Done.

File: driver/utils/utils.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (traceon) [reply driver/utils/utils.h]: Done.

File: driver/utils/utils.h [hunk 5]
```diff
@@ -179,8 +181,35 @@ inline SQLRETURN fillOutputBuffer(
 
     const auto bytes_to_copy = std::min<std::int64_t>(in_value_length, out_value_max_length);
     std::memcpy(out_value, in_value, bytes_to_copy);
+    
+    if (bytes_to_copy > out_value_max_length)
+        return SQL_SUCCESS_WITH_INFO;
```
Comment (Enmk) [inline driver/utils/utils.h]: Why not just throw an exception here?

Comment (traceon) [reply driver/utils/utils.h]: Some processing later done in `fillOutputString()` (namely, null-terminating, if requested), so I would need to try-catch-retrhow in that case. That would be both slow, and hard to read/understand/follow code.

Comment (traceon) [reply driver/utils/utils.h]: (Even when truncated, strings have to be null-terminated.)

File: driver/utils/utils.h (hunk 4 above)
[the remark this answers is not in the document]
Comment (traceon) [reply driver/utils/utils.h]: Done.

**Pull request merged**